document runtime and shim configuration and selection (#9145)
containerd/containerd: docs/PLUGINS.md

### V2 Runtimes

containerd supports multiple container runtimes. Each container can be

> invoked with a different runtime. The runtimes can be configured in the

Reviewer: ```suggestion
invoked with a different runtime. For container runtime interface (CRI) integrations the runtimes can be configured in the
```

Author: Added

> containerd configuration file, including setting a default runtime,

Author: Is there any other way? I thought under v2, all of the runtimes are listed solely in that section of the config?

> containerd configuration file, including setting a default runtime,

Reviewer 2: containerd itself only has a single hard-coded default per OS: https://github.com/containerd/containerd/blob/b30e0163ac36c1a193604e5eca031053d62019c5/defaults/defaults_linux.go#L21

When a client (such as `ctr` or `nerdctl`) attempts to run a container, it will either use that hard-coded default or specify the shim as an option.

The CRI plugin (which is effectively another client) has its own configuration section which can specify other defaults.

> containerd configuration file, including setting a default runtime,

Reviewer: pls include Sam's text here "When a client (such as ctr or nerdctl) attempts to run a container, it will either use that hard-coded default or specify the runtime shim as an option."

> or can be specified explicitly when creating a container via gRPC.

Reviewer: ```suggestion
or can be specified explicitly when creating a container via CRI gRPC by selecting the runtime handler to be used selected from the runtimes configured for CRI users.
```

Author: Added

### V2 Runtimes

> containerd supports multiple container runtimes. Each container can be
> invoked with a different runtime. For container runtime interface (CRI) integrations the
> runtimes can be configured in the
> containerd configuration file, including setting a default runtime,
> or can be specified explicitly when creating a container via CRI gRPC
> by selecting the runtime handler to be used selected from the runtimes configured for CRI users.

Reviewer 2: Trying to rewrite a bit for flow:

```suggestion
containerd supports multiple container runtimes. Each container can be
invoked with a different runtime.

When using the CRI (Container Runtime Interface) plugin, named runtimes can be defined in the
containerd configuration file. When a container is run without specifying a runtime, the configured default runtime is used.  Alternately, a different named runtime can be specified explicitly when creating a container via CRI gRPC
by selecting the runtime handler to be used.
```

Author: I like it. Done.

> When a client, such as `ctr` or `nerdctl`, requests of containerd to run a container, the client either will specify the specific
> runtime to use, along with options desired, or, if none is specified, containerd will use that hard-coded default.

Reviewer 2: ```suggestion
When a client such as `ctr` or `nerdctl` creates a container, it can optionally specify a runtime and options to use.  If a runtime is not specified, containerd will use its default runtime.
```

Author: Done

> The runtime v2 interface allows resolving defining specific runtimes as binaries on the system,
> which are used to start the shim process for containerd. This, in turn, allows

Reviewer 2: ```suggestion
containerd invokes v2 runtimes as binaries on the system,
which are used to start the shim process for containerd. This, in turn, allows
```

Author: Done
